### PR TITLE
Fix data race

### PR DESCRIPTION
**Pull request by longjiquan**
fix: #23125

### CODE REVIEW

File: internal/querycoordv2/task/task.go [hunk 1]
```diff
@@ -95,6 +96,7 @@ type baseTask struct {
 	status   *atomic.Int32
 	priority Priority
 	err      error
+	errMu    sync.RWMutex // seems not good.
```
Comment (jiaoew1991) [inline internal/querycoordv2/task/task.go:99]: use atomic ? It looks simpler

Comment (longjiquan) [reply internal/querycoordv2/task/task.go:99]: OK, I'll fix this.

**Pull request merged**